src/utils.ts

```diff
@@ -180,16 +180,36 @@ export function encodeInputArgsForViewRequest(type: string, value: string) {
 }
 
 // Deserialize "[1,2,3]" or "1,2,3" to ["1", "2", "3"]
-export function deserializeVector(vectorString: string): string[] {
-  let result = vectorString.trim();
-  if (result[0] === "[" && result[result.length - 1] === "]") {
-    result = result.slice(1, -1);
+// Also works nested [[1, 2], [3, 4]] to [["1", "2"], ["3", "4"]]
+// TODO: Need to type this recursively
+export function deserializeVector(vectorString: string): any {
+  const trimmed = vectorString.trim();
+  let inside = "";
+  if (isArrayStr(trimmed)) {
+    inside = trimmed.slice(1, -1);
   }
   // There's a tradeoff here between empty string, and empty array.  We're going with empty array.
-  if (result.length == 0) {
+  if (inside.length == 0) {
     return [];
+  } else if (isArrayStr(inside)) {
+    const innerArrayStrs = [];
+    let build = "";
+    for (let i = 0; i < inside.length; i++) {
+      if (inside[i] === "]") {
+        innerArrayStrs.push(build);
+        build = "";
+      } else {
+        build += inside[i];
+      }
+    }
+    return innerArrayStrs.map(deserializeVector);
```

Comment:
This code will have the bracket at the beginning?

But, I guess you call it recursively?

I think this misses, can we make a unit test, I suspect that:

```
[[blah,blah],[blah,blah]]
```
Will get to:
```
[blah, blah], [blah, blah]
```
Then 
```
innerArrayStrs.push('[blah, blah')
innerArrayStrs.push(',[blah, blah')
```

Then do that recursively which will return two items

```
'[blah', 'blah'
'','[blah', 'blah'
```